Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1062,7 +1062,20 @@ extension SwiftLanguageServer {
   }
 
   public func documentDiagnostic(_ req: DocumentDiagnosticsRequest) async throws -> DocumentDiagnosticReport {
-    return try await .full(fullDocumentDiagnosticReport(req))
+    do {
+      return try await .full(fullDocumentDiagnosticReport(req))
+    } catch {
+      // VS Code does not request diagnostics again for a document if the diagnostics request failed. 
+      // Since sourcekit-lsp usually recovers from failures (e.g. after sourcekitd crashes), this is undesirable. 
+      // Instead of returning an error, return empty results.
```

Comment:
Is this a VS Code thing or an extension thing? Because IMO it's useful to distinguish between having results and not (consider eg. deciding whether to replace previous issues or not).

Reply:
AFAIK it’s a VS Code thing. @adam-fowler to confirm.

Reply:
This is a VSCode thing. The extension doesn't have control over when VSCode asks for diagnostics. 

Out of interest the "failed" diagnostics request, does that mean no response was returned or that an error response was returned.

Reply:
By “failed” I meant an error response.